packages/expo-errors/src/warnDeprecated.ts

```diff
@@ -0,0 +1,50 @@
+import semver from 'semver';
+import Constants from 'expo-constants';
+
+const postedWarnings: { [key: string]: boolean } = {};
+
+/**
+ * Used for deprecating values and throwing an error if a given version of Expo has passed.
+ */
+export default function warnDeprecated(
+  library: string,
+  deprecated: string,
+  options: {
+    replacement?: string;
+    versionToRemove?: string;
+  } = {}
+) {
+  const { versionToRemove, replacement } = options;
+  const key = `${library}:${versionToRemove}:${deprecated}:${replacement}`;
+  if (!postedWarnings[key]) {
+    postedWarnings[key] = true;
+  }
+
+  if (
+    !versionToRemove ||
+    !versionToRemove.length ||
```

Comment:
The .length checks are mostly harmless but I don't think they are needed either.